master/internal/rm/kubernetesrm/pods.go

```diff
@@ -448,6 +468,30 @@ func (p *pods) reattachAllocationPods(ctx *actor.Context, msg reattachAllocation
 		return nil
 	}
 
+	// This is needed to label pods created before Determined supported k8s agent enable disable.
+	// We will not reattach pods that are queued and don't have the affinity that respects
+	// agent disabling. Not many people should be relying on this feature when this will be released
+	// since it was behind _agent_reattach_enabled until the version this is also released on.
+	// We can't patch the pods with the needed field, as a limitation of Kubernetes.
+	for _, pod := range k8sPods {
+		pod := pod
+		if pod.Spec.NodeName == "" { // Only do this for pods not assigned to a node yet.
+			before := pod.DeepCopy()
+			addNodeDisabledAffinityToPodSpec(pod, clusterIDNodeLabel())
+
+			if !reflect.DeepEqual(pod.Spec, before.Spec) {
+				p.deleteKubernetesResources(ctx, pods, configMaps)
+				ctx.Respond(fmt.Errorf(
+					"unable to restore pod %s since it was queued and does not have "+
+						"Determined's affinity to prevent scheduling on disabled nodes. "+
+						"This is expected to happen on allocations with queued pods "+
+						"when upgrading from before or equal to 0.25.1 "+
+						"to after or equal to 0.26.0", pod.Name))
+				return nil
+			}
+		}
+	}
```

Comment:
nit, pull this into a function with a name that makes it obvious you don't need to think about it ever again after it is written.